File: src/python/contains_duplicate_ii.py
#!/usr/bin/python

class Solution:
    # @param {integer[]} nums
    # @param {integer} k
    # @return {boolean}
    def containsNearbyDuplicate(self, nums, k):
        dict = {}
        for i in range(len(nums)):
            if dict.has_key(nums[i]):
                firIdx = dict.get(nums[i])
                if i - firIdx <= k:
                    return True
            dict[nums[i]] = i

        return False

File: src/ContainsDuplicateII.java
import java.util.HashMap;

public class ContainsDuplicateII {
	public boolean containsNearbyDuplicate(int[] nums, int k) {
		HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
		
		for (int i = 0; i < nums.length; i ++) {
			if (map.containsKey(nums[i])) {
				int firIdx = map.get(nums[i]);
				if (i - firIdx <= k) {
					return true;
				} else {
					map.put(nums[i], i);
				}
			} else {
				map.put(nums[i], i);
			}
		}
		
		return false;
	}
}
